### DISCUSSION TITLE
Voisiko Voikko-oikoluvun integroida Mozilla Thunderbirdiin?

### DISCUSSION THREAD

**Discussion by teafusion** [voikko/discussion#15] [Ideas]
Nyt kun Windows Mail on mennyttä ja korvataan uudella Outlookilla, Thunderbird oletettavasti kasvattaa suosiota ainakin jonkin verran, joten olisi mukavaa, jos Thunderbird saisi käyttöliittymän suomi-kielipaketin lisäksi myös suomalaisen oikoluvun ja Voikkohan olisi siihen loistava komponentti, eikö vaan? Thunderbird on mainio avoimen lähdekoodin vastine Outlookille, joten sen toivoisi saavan suosiota, myös suomenkielisten käyttäjien keskuudessa, jota oikoluku auttaisi merkittävästi. Voisiko integrointi Firefoxiinkin olla mahdollista samalla?

Kiitos kaikille Voikko-oikolukua kehittäville ja ylläpitäville koodareille, käytän sitä LibreOfficessa, sen minkä tekstinkäsittelyä tarvitsen, toisaalta sähköposti on huomattavasti useammin itsellä käytössä silloin, kun tekstiä tulee tuotettua.

**Comment by bigjtu** [comment]
Vuosia, vuosia sitten Voikko olikin tuettu Mozillan tuotteissa (eli Thunderbirdissä), mutta ilmeisesti Mozillan teknisten muutosten takia tuki hiipui. Koska en ole koodari, en teknisesti osaa ollenkaan sanoa, kuinka iso savotta olisi palauttaa tuo tuki - saati ylläpitää sitä - mutta jos joku sen päättäisi ottaa niskoilleen, osallistuisin mielelläni kolehtiin.

**Comment by iikka81** [comment]
Toivottavasti joku ottaisi tämän tehtäväkseen. Meillä töissä käytetään Thunderbirdiä ja suomen oikoluku olisi ihan kiva olla.

**Comment by tjyrinki** [comment]
Mozillan laajennuksista huomaan että tuollainen kehittäjän itsensä mukaan "Voikkoa huonompi" on olemassa https://addons.mozilla.org/fi/firefox/addon/finnish-spellchecker-dict/, josta sitten viittaus on saman kehittäjän kokeelliseen, "ei toimi joka sivulla mutta oikoluku parempi" Voikkoa käyttävään laajennukseen https://addons.mozilla.org/en-US/firefox/addon/fin-spell/

Tuota kokeellista ei ole kehitetty sitten vuoden 2021 jälkeen - itse koodi löytyy näemmä https://github.com/fluks/fin-spell - mutta siinä olisi pohja josta jatkaa jos pysyy koodi hyppysissä eikä Mozilla laajennoksia ole vuoden 2021 jälkeen entisestään rajoitettu.

**Reply by user-1-2-3-4-5** [reply]
Miksi Firefoxin oikoluku on toteutettu lisäosalla [sanakirjapaketin](https://addons.mozilla.org/en-GB/firefox/language-tools/) sijaan?

Tosin koska näin toimi myös [edesmennyt Voikon oma lisäosa](https://voikko.puimula.org/sovellukset.html), lienee sanakirja rajoittaisi Voikon toiminnallisuutta. Kumma tosin, ettei pelkkää sanalistaa ole käännettu ja tuotu Firefoxiin sekä Thunderbirdiin (myöskään kolmannen osapuolen toimesta).

**Reply by tjyrinki** [reply]
Syynä tähän on se mikä alunperinkin sai Voikon liikkeelle - suomen kieli ei yksinkertaisesti taivu sanakirjapaketiksi. Tai siis vitsaillen taipuu liiaksi, jotta mikään sanakirjapaketti ei voi sitä kunnolla kattaa. Toki jos Mozilla ei koskaan aio tukea tällaisia kieliä kunnolla, voisi joku yrittää tukia uusimpia myspell/hunspell-ominaisuuksia (tai mitä kirjasto tarkalleen ottaen Mozilla nykyään käyttääkin), voiko nykyään saada hieman vähemmän huonosti toimivan sanakirjapaketin aikaiseksi kuin aiemmin. Eli tuetaanko taivutusmuotoja millä tavoin, onko jotain kehitystä tapahtunut viimeisen 10+ vuoden aikana.

**Comment by user-1-2-3-4-5** [comment]
Loin [oman aiheen Firefoxille](https://github.com/orgs/voikko/discussions/52) (en huomannut irrallista mainintaasi aloitusviestissä).
